dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java

```diff
@@ -320,6 +320,30 @@ public Set<String> getMyOrganizations() {
         }
     }
 
+    /**
+     * Determine if the specified organization is one that github considers the user has "access" to.
+     * Incorporates some optimizations for the common cases, and falls back to getMyOrganizations()
+     */
+    public boolean isOneOfMyOrganizations(String organization) {
+        try {
+            // The following statements could be condensed into a single OR conditional, but it would be hard to comment, and you would have to grok the short-circuiting to understand it...
+            // The user can always access their own account.
+            if (organization.equals(githubTokenUsername)) {
+                return true;
+            }
+            // Check the list of organizations that github says the user can "access".
+            // Note that getAllOrganizations() only returns organizations, not personal user accounts.
+            if (github.getMyself().getAllOrganizations().stream().anyMatch(org -> org.getLogin().equals(organization))) {
```

Comment:
So, I think we're using the `getAllOrganizations` and `listRepositories` calls because in addition to membership, they include orgs/repos that the user has "access" to, which at a minimum includes orgs the user (and the user's teams, I think) collaborates with.  If we just wanna check if a user is a member of an org, then yes, the check is much more simple and linear.